File: visualiser/facades/ociVault.py
#!/usr/bin/python

# Copyright (c) 2020, 2024, Oracle and/or its affiliates.
# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.

"""Provide Module Description
"""

# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#
__author__ = ["Andrew Hopkinson (Oracle Cloud Solutions A-Team)"]
__version__ = "1.0.0"
__module__ = "ociVault"
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#


import oci

from common.okitLogging import getLogger
from facades.ociConnection import OCIVaultConnection

# Configure logging
logger = getLogger()


class OCIVaults(OCIVaultConnection):
    def __init__(self, config=None, configfile=None, profile=None, compartment_id=None):
        self.compartment_id = compartment_id
        self.vaults_json = []
        super(OCIVaults, self).__init__(config=config, configfile=configfile, profile=profile)

    def list(self, compartment_id=None, filter=None):
        if compartment_id is None:
            compartment_id = self.compartment_id

        # Add filter to only return AVAILABLE Compartments
        if filter is None:
            filter = {}

        if 'lifecycle_state' not in filter:
            filter['lifecycle_state'] = 'AVAILABLE'

        vaults = oci.pagination.list_call_get_all_results(self.client.list_vaults, compartment_id=compartment_id).data

        # Convert to Json object
        vaults_json = self.toJson(vaults)
        logger.debug(str(vaults_json))

        # Filter results
        self.vaults_json = self.filterJsonObjectList(vaults_json, filter)
        logger.debug(str(self.vaults_json))

        return self.vaults_json

File: ocd/packages/react/src/components/properties/provider/oci/resources/configs/OciVault.ts
/*
** Copyright (c) 2020, 2024, Oracle and/or its affiliates.
** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
*/

import { ResourceElementConfig } from "../../../../OcdPropertyTypes"
import { OciCommonConfigs } from "../../OciCommonConfigs"

export namespace OciVaultConfigs {
    export function configs(): ResourceElementConfig[] {
        return [
            ...OciCommonConfigs.configs(),
            {
                id: 'vault_type',
                properties: {},
                configs: [],
                options: [
                    {id: 'DEFAULT', displayName: 'Default'},
                    {id: 'VIRTUAL_PRIVATE', displayName: 'Virtual Private Vault'},
                ]
            },
        ]
    }
}
